job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -944,17 +944,17 @@ class PublisherContext extends AbstractExtensibleContext {
      *
      * @since 1.23
      */
-    @RequiresPlugin(id = 'stashNotifier')
+    @RequiresPlugin(id = 'stashNotifier', minimumVersion = '1.8')
```

Comment:
This would break compatibility for everyone using an older version. Before increasing the minimum version, support for older versions must be deprecated. Add a call to logPluginDeprecationWarning (e.g. https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.37/job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy#L746), add an entry to the migration page (https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.37/docs/Migration.md) and add an entry to the release notes (https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.37/docs/Home.md).